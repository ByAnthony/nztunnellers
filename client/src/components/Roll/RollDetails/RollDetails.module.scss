@use 'src/index.scss';

.tunneller {
  flex: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: index.$m-margin;

  @include index.primary-button-link;

  @include index.desktop {
    padding: index.$padding-card;
    flex: 50%;
  }

  p {
    text-align: left;
  }

  .surname {
    margin: 0;
    font-size: 1.5rem;
    font-weight: 500;
    color: index.$secondary-color;

    @include index.tablet {
      font-size: 1.75rem;
    }

    @include index.desktop {
      font-size: 1.8rem;
    }
  }

  .forename {
    margin: 0;
    font-size: 1.5rem;
    font-weight: 200;
    color: index.$secondary-color;

    @include index.tablet {
      font-size: 1.75rem;
    }

    @include index.desktop {
      font-size: 1.8rem;
    }
  }

  .dates {
    margin: index.$xs-margin 0 0 0;
    font-size: 1rem;
    font-weight: 200;
    letter-spacing: 2px;
    color: index.$secondary-color;

    @include index.tablet {
      margin-top: index.$s-margin;
    }

    @include index.desktop {
      margin-top: index.$m-margin;
      font-size: 1.125rem;
    }
  }
}
