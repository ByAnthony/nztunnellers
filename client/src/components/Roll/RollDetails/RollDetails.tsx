import { displayBiographyDates } from '../../../utils/displayBiographyDates';

import { Details } from '../../../types/roll';

import STYLES from './RollDetails.module.scss';

type Props = {
  listOfTunnellers: Details[];
}

export function RollDetails({ listOfTunnellers }: Props) {
  return (
    <>
      {listOfTunnellers.map((tunneller: Details) => (
        <a href={`/tunnellers/${tunneller.id}`} key={tunneller.id}>
          <div className={STYLES.tunneller}>
            <div>
              <p className={STYLES.forename}>{tunneller.name.forename}</p>
              <p className={STYLES.surname}>{tunneller.name.surname}</p>
              <p className={STYLES.dates}>
                { displayBiographyDates(tunneller.birth, tunneller.death) }
              </p>
            </div>
            <div className={STYLES.arrow}>&rarr;</div>
          </div>
        </a>
      ))}
    </>
  );
}
