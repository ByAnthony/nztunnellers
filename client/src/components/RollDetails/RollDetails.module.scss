@use 'src/index.scss';

a {
  color: index.$tertiary-color;
  text-decoration: none;
}

.tunneller {
  flex: 100%;
  padding: 25px 25px 22px 25px;
  font-stretch: 85%;
  letter-spacing: 1px;
  text-align: left;
  border: 1px solid index.$secondary-color;
  border-radius: index.$border-radius;
  transition: border-color 0.9s ease-in-out, border-color 0.9s ease-in-out;
  list-style: none;
  cursor: pointer;

  @include index.tablet {
    padding: 30px 30px 27px 30px;
  }

  @include index.desktop {
    padding: 40px 40px 35px 40px;
    flex: 50%;
  }

  &:hover {
    border: 1px solid index.$tertiary-color;
  }
}

.surname {
  margin: 0;
  font-size: 1.875rem;
  font-weight: 700;

  @include index.tablet {
    font-size: 2.5rem;
  }

  @include index.desktop {
    font-size: 2.8125rem;
  }
}

.forename {
  margin: 0;
  font-size: 1.375rem;
  font-weight: 300;

  @include index.tablet {
    font-size: 1.75rem;
  }

  @include index.desktop {
    font-size: 1.875rem;
  }
}

.dates {
  margin: 15px 0 0 0;
  font-size: 1rem;
  font-weight: 200;
  letter-spacing: 2px;

  @include index.tablet {
    margin-top: 20px;
    font-size: 1.125rem;
  }

  @include index.desktop {
    margin-top: 25px;
    font-size: 1.125rem;
  }
}
