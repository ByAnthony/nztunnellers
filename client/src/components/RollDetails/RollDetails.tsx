import { Link } from 'react-router-dom';
import { Details } from '../../types/roll';
import STYLES from './RollDetails.module.scss';

type Roll = {
    listOfTunnellers: Details[];
}

export function RollDetails({ listOfTunnellers }: Roll) {
  return (
    <>
      {listOfTunnellers.map((tunneller: Details) => (
        <Link to={`/roll/${tunneller.id}`} key={tunneller.id}>
          <div className={STYLES.tunneller}>
            <p className={STYLES.surname}>{tunneller.name.surname}</p>
            <p className={STYLES.forename}>{tunneller.name.forename}</p>
            <p className={STYLES.serial}>{tunneller.serial}</p>
          </div>
        </Link>
      ))}
    </>
  );
}
