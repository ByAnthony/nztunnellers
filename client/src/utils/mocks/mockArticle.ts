import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
