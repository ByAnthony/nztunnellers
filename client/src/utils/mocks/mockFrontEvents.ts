import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
