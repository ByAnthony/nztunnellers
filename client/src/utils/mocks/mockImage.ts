import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'Courtesy of John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook: ImageBook = {
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: 'p. 89',
};

export const mockImageSource: ImageSource = {
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
};

export const mockImage: Image = {
  url: '1/1234.jpg',
  source: mockImageSource,
};
