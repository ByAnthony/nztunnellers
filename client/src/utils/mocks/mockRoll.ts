import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails = {
  id: mockId,
  name: johnDoe,
  birth: '1886',
  death: '1952',
};

export const mockRoll = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
