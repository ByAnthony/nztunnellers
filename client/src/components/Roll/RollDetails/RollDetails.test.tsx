import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import { mockDetailsList } from '../../../utils/mocks/mockRoll';

import { RollDetails } from './RollDetails';

const component = <RollDetails listOfTunnellers={mockDetailsList} />;

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('renders details', () => {
  render(component);

  expect(screen.getByText('John')).toBeInTheDocument();
  expect(screen.getByText('Doe')).toBeInTheDocument();
  expect(screen.getByRole('link', { name: 'John Doe 1886-1952 →' }).getAttribute('href')).toBe('/tunnellers/26');

  expect(screen.getByText('Alexander')).toBeInTheDocument();
  expect(screen.getByText('Driscott')).toBeInTheDocument();
  expect(screen.getByRole('link', { name: 'Alexander Driscott 1886-1952 →' }).getAttribute('href')).toBe('/tunnellers/1');
});
