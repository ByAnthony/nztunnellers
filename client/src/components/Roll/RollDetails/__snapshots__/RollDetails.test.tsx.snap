// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <a
    href="/tunnellers/26"
  >
    <div
      class="tunneller"
    >
      <div>
        <p
          class="forename"
        >
          John
        </p>
        <p
          class="surname"
        >
          Doe
        </p>
        <p
          class="dates"
        >
          1886-1952
        </p>
      </div>
      <div
        class="arrow"
      >
        →
      </div>
    </div>
  </a>
  <a
    href="/tunnellers/1"
  >
    <div
      class="tunneller"
    >
      <div>
        <p
          class="forename"
        >
          Alexander
        </p>
        <p
          class="surname"
        >
          Driscott
        </p>
        <p
          class="dates"
        >
          1886-1952
        </p>
      </div>
      <div
        class="arrow"
      >
        →
      </div>
    </div>
  </a>
</DocumentFragment>
`;
